src/Symfony/Component/Console/Helper/SymfonyQuestionHelper.php

```diff
@@ -39,6 +41,8 @@ public function ask(InputInterface $input, OutputInterface $output, Question $qu
             } else {
                 // make required
                 if (!is_array($value) && !is_bool($value) && 0 === strlen($value)) {
+                    @trigger_error('The default question validator is deprecated since Symfony 3.4 and will not be used anymore in version 4.0. Set a custom question validator instead.', E_USER_DEPRECATED);
```

Comment:
We're deprecating the fact this exception is thrown when the value is blank. Meaning in 4.0 ask() is optional by default, whereas now it's required by default.

I actually considered putting the trigger above `if (!is_array($value) && !is_bool($value) && 0 === strlen($value)) {`, but its weird to get the deprecation for a case that doesnt change in 4.0.